python-sdk/src/astro/databases/snowflake.py

```diff
@@ -622,13 +645,48 @@ def load_file_to_table_natively(
         )
         self.evaluate_results(rows)
 
+    def _get_copy_into_with_metadata_sql_statement(
```

Comment:
Not a blocker, but would be nice to split this into multiple functions.

Reply:
+1